pkg/operators/kubenetworkmanager/kubenetworkmanager.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubenetworkmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/kubemanager"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+// TODO: Where should this be defined?
+type RemoteKind string
+
+const (
+	RemoteKindPod     RemoteKind = "pod"
+	RemoteKindService RemoteKind = "svc"
+	RemoteKindOther   RemoteKind = "other"
+)
+
+// TODO: More granualar and "shareable" interfaces
```

Comment:
What about defining some types that gadgets can include to represent a network enpdoint? 

```golang
type Endpoint struct {
        Addr string // IP address
        Kind RemoteKind 
        Name string // pod name or service name (even domain name?)
        Namespace string // valid when Kind == Pod || Kind == Service
        PodLabels map[string]string // only of Kind == Pod 
        PodHostIP string
}
```

Then, it gadget could use it in their types representation

```golang 
...

Local types.Endpoint
Remote types.Endpoint
...
```

And there could a a SetEndpoint interface:

```golang
SetRemoteEndpoint(*Endpoint)
SetLocalEndpoint(*Endpoint)

GetLocalIP()?
GetRemoteIP()?
```

However, it seems to me it's related to #996, so I'd prefer to leave this interface as it is and handle all of this later on there. WDYT?

Reply:
I agree that we should do it in a later stage. The `Endpoint struct` looks nice too.
What we have to have in mind is that resolving the Endpoint information is done different for the `LocalIP` and `RemoteIP` (at least in this gadget)

In `trace network` we resolve all Local-Endpoint information through the Pod-Name and Pod-Namespace. All Remote-Endpoint information is currently only relying on the capture remote IP from `trace network`